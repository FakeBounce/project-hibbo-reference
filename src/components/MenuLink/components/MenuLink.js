import React from 'react';
import { func } from 'prop-types';

import TouchableRipple from '../../shared/TouchableRipple';
import TextView from './TextView';

const MenuLink = ({ onBeginPress, onStopPress, to, ...props }) => {
  return (
    <TouchableRipple
      onPressIn={onBeginPress}
      onPressOut={onStopPress}
      onPress={to}
    >
      <TextView {...props} />
    </TouchableRipple>
  );
};

MenuLink.propTypes = {
  onBeginPress: func.isRequired,
  onStopPress: func.isRequired,
  to: func.isRequired,
};

export default MenuLink;
