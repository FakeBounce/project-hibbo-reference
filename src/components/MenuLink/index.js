import React, { PureComponent } from 'react';
import { Animated } from 'react-native';
import { connect } from 'react-redux';

import { bind } from 'decko';

import MenuLink from './components/MenuLink';

const mapStateToProps = state => {
  return {
    device: state.device,
  };
};

class MenuLinkContainer extends PureComponent {
  constructor(props) {
    super(props);

    // console.log(this.props.device);

    const animationDelay = 300;

    this.linkOpacity = new Animated.Value(1);
    this.linkScale = new Animated.Value(1);

    this.onBeginPressAnimation = Animated.parallel([
      Animated.timing(this.linkOpacity, { toValue: 0.5, duration: animationDelay }),
      Animated.timing(this.linkScale, { toValue: 1.3, duration: animationDelay }),
    ]);

    this.onStopPressAnimation = Animated.parallel([
      Animated.timing(this.linkOpacity, { toValue: 1, duration: animationDelay }),
      Animated.timing(this.linkScale, { toValue: 1, duration: animationDelay }),
    ]);
  }

  @bind
  onBeginPress() {
    this.onBeginPressAnimation.start();
  }

  @bind
  onStopPress() {
    this.onStopPressAnimation.start();
  }

  render() {
    return (
      <MenuLink
        {...this.props}
        onBeginPress={this.onBeginPress}
        onStopPress={this.onStopPress}
        linkScale={this.linkScale}
        linkOpacity={this.linkOpacity}
      />
    );
  }
}

export default connect(mapStateToProps)(MenuLinkContainer);
