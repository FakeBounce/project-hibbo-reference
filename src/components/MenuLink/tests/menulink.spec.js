import React from 'react';
import { Animated } from 'react-native';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

import MenuLink from '../components/MenuLink';
import MenuLinkContainer from '../';

describe('MenuLink', () => {
  const text = 'test';
  const linkScale = new Animated.Value(1);
  const linkOpacity = new Animated.Value(1);
  const onBeginPress = jest.fn();
  const onStopPress = jest.fn();
  const to = jest.fn();

  const createMenuLink = device => {
    return renderer
      .create(
        <MenuLink
          text={text}
          linkScale={linkScale}
          linkOpacity={linkOpacity}
          device={device}
          onBeginPress={onBeginPress}
          onStopPress={onStopPress}
          to={to}
        />,
      )
      .toJSON();
  };

  const createMenuLinkContainer = device => {
    const mockStore = configureStore();
    const store = mockStore({
      device,
    });
    return renderer
      .create(
        <Provider store={store}>
          <MenuLinkContainer
            text={text}
            linkScale={linkScale}
            linkOpacity={linkOpacity}
            device={device}
            onBeginPress={onBeginPress}
            onStopPress={onStopPress}
            to={to}
          />
        </Provider>,
      )
      .toJSON();
  };

  it('should render the component for iphone', () => {
    const menuLink = createMenuLink('iphone');

    expect(menuLink).toMatchSnapshot();
  });

  it('should render the component for ipad', () => {
    const menuLink = createMenuLink('ipad');

    expect(menuLink).toMatchSnapshot();
  });

  it('should render the container for iphone', () => {
    const menuLink = createMenuLinkContainer('iphone');

    expect(menuLink).toMatchSnapshot();
  });

  it('should render the container for ipad', () => {
    const menuLink = createMenuLinkContainer('ipad');

    expect(menuLink).toMatchSnapshot();
  });

  it('should click on menu', () => {
    const tree = shallow(
      <MenuLink
        text={text}
        linkScale={linkScale}
        linkOpacity={linkOpacity}
        onBeginPress={onBeginPress}
        onStopPress={onStopPress}
        to={to}
      />,
    );

    tree.find('TouchableRipple').first().simulate('pressIn');
    tree.find('TouchableRipple').first().simulate('press');
    tree.find('TouchableRipple').first().simulate('pressOut');
    expect(to).toHaveBeenCalled();
    expect(onBeginPress).toHaveBeenCalled();
    expect(onStopPress).toHaveBeenCalled();
  });

  it('should click on menu container', () => {
    const mockStore = configureStore();
    const store = mockStore({
      device: 'iphone',
    });
    const tree = shallow(
      <Provider store={store}>
        <MenuLinkContainer
          text={text}
          linkScale={linkScale}
          linkOpacity={linkOpacity}
          onBeginPress={onBeginPress}
          onStopPress={onStopPress}
          to={to}
        />
      </Provider>,
    );

    tree.find('TouchableRipple').first().simulate('pressIn');
    tree.find('TouchableRipple').first().simulate('press');
    tree.find('TouchableRipple').first().simulate('pressOut');
    expect(to).toHaveBeenCalled();
    expect(onBeginPress).toHaveBeenCalled();
    expect(onStopPress).toHaveBeenCalled();
  });
});
