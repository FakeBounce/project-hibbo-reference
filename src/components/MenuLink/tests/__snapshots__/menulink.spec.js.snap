// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLink should render the component for ipad 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        collapsable={undefined}
        style={
          Object {
            "height": 100,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Object {},
            ]
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`MenuLink should render the component for iphone 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        collapsable={undefined}
        style={
          Object {
            "height": 62,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Object {},
            ]
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`MenuLink should render the container for ipad 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        collapsable={undefined}
        style={
          Object {
            "height": 100,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Object {},
            ]
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`MenuLink should render the container for iphone 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        collapsable={undefined}
        style={
          Object {
            "height": 62,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Object {},
            ]
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;
